pkg/cpi/commonUtils.go

```diff
@@ -91,3 +107,39 @@ func (tokenParameters TokenParameters) GetBearerToken() (string, error) {
 	token := jsonResponse.Path("access_token").Data().(string)
 	return token, nil
 }
+
+// HandleHTTPFileDownloadResponse - handle the file download response for http multipart response
+func (httpFileDownloadRequestParameters HttpFileDownloadRequestParameters) HandleHTTPFileDownloadResponse() error {
+	response := httpFileDownloadRequestParameters.Response
+	contentDisposition := response.Header.Get("Content-Disposition")
+	disposition, params, err := mime.ParseMediaType(contentDisposition)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read filename from http response headers, Content-Disposition "+disposition)
+	}
+	filename := params["filename"]
+
+	if response != nil && response.Body != nil {
+		defer response.Body.Close()
+	}
+
+	if response.StatusCode == 200 {
+		workspaceRelativePath := httpFileDownloadRequestParameters.FileDownloadPath
+		err = os.MkdirAll(workspaceRelativePath, 0755)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to create workspace directory")
+		}
+		zipFileName := filepath.Join(workspaceRelativePath, filename)
+		file, err := os.Create(zipFileName)
+		if err != nil {
+			return errors.Wrapf(err, httpFileDownloadRequestParameters.ErrMessage)
```

Comment:
its just struct holding input arguments for HandleHTTPFileDownloadResponse() method

Reply:
And this is turn holds information why `os.Create(zipFileName)` failed?

Reply:
fixed

Reply:
Could you reference the commit, because I don't see a change. And I'm not sure there needs to be a change. It was a question, not a suggestion.

Reply:
Now it is gone. Maybe it would still be interesting to see the `httpFileDownloadRequestParameters.ErrMessage` in some cases. For example, if the ErrMessage field exists then fail and print the message to the log.

Reply:
I see it is being returned with the response code check.